spotguide/spotguide.go

```diff
@@ -132,44 +126,112 @@ func (r LaunchRequest) RepoFullname() string {
 	return r.RepoOrganization + "/" + r.RepoName
 }
 
-func downloadGithubFile(githubClient *github.Client, owner, repo, file, tag string) ([]byte, error) {
-	reader, err := githubClient.Repositories.DownloadContents(ctx, owner, repo, file, &github.RepositoryContentGetOptions{
-		Ref: tag,
-	})
+type ReleaseBody struct {
+	Pipeline string `json:"pipeline" yaml:"pipeline"`
+}
+
+// SpotguideManager is responsible to scrape spotguides on GitHub and persist them to database
+type SpotguideManager struct {
+	pipelineVersion       *semver.Version
+	githubToken           string
+	spotguideOrganization *auth.Organization
+}
+
+func NewSpotguideManager(pipelineVersionString string) *SpotguideManager {
+	githubToken := viper.GetString("github.token")
+
+	spotguideOrganization, err := auth.GetOrganizationByName(SpotguideGithubOrganization)
 	if err != nil {
-		return nil, err
+		log.Errorf("shared spotguide organization (%s) is not found", SpotguideGithubOrganization)
 	}
 
-	defer reader.Close()
+	pipelineVersion, _ := semver.NewVersion(pipelineVersionString)
 
-	return ioutil.ReadAll(reader)
-}
-
-func internalScrapeSpotguides(orgID uint, userID uint) {
-	if err := ScrapeSpotguides(orgID, userID); err != nil {
-		log.Warnf("failed to scrape Spotguide repositories for org [%d]: %s", orgID, err)
+	s := &SpotguideManager{
+		pipelineVersion:       pipelineVersion,
+		spotguideOrganization: spotguideOrganization,
+		githubToken:           githubToken,
 	}
+
+	// subscribe to organization creations and sync spotguides into the newly created organizations
+	authEventEmitter.NotifyOrganizationRegistered(s.internalScrapeSpotguides)
```

Comment:
Register this subscriber outside of this method (this is not a SpotguideManager responsibility)